### PR TITLE
Initial preparation for v6.0

### PR DESCRIPTION
**Pull request by dbhynds**
## Summary of Changes

Marks several objects for deprecation in the next major release. All of these methods are from the original library before we took it over. They don't follow modern PHP paradigms.

Specifically, this deprecates the old IMSGlobal objects. It also removes several functions that render HTML content.

This is blocked by https://github.com/packbackbooks/lti-1-3-php-library/pull/99

## Testing

- [ ] I have added automated tests for my changes
- [ ] I ran `composer test` before opening this PR
- [ ] I ran `composer lint-fix` before opening this PR

### CODE REVIEW

**Comment by lin-brian-l** [comment]
@dbhynds sorry, I'm lacking some context - can you explain what we're going to change in v6.0 and why we can remove those functions?

File: src/ImsStorage/ImsCache.php [hunk 1]
```diff
@@ -4,6 +4,9 @@
 
 use Packback\Lti1p3\Interfaces\ICache;
 
+/**
+ * @deprecated
+ */
```
Comment (dbhynds) [inline src/ImsStorage/ImsCache.php:9]: Both of these classes are leftover from the old library. I left them in originally as examples, but frankly, they're a mess and won't really function in a modern web app. I'd rather leave it up to people to create their own implementations.

File: src/JwksEndpoint.php [hunk 2]
```diff
@@ -48,8 +48,13 @@ public function getPublicJwks()
         return ['keys' => $jwks];
     }
 
+    /**
+     * @deprecated
```
Comment (dbhynds) [inline src/JwksEndpoint.php:52]: It's really just a bad idea in general to `echo` anything out in a modern php app. If someone needs to echo out the json payload, it should be easy enough for them to just write the code on line 58 themselves in their app.

File: src/LtiDeepLink.php [hunk 3]
```diff
@@ -44,17 +44,12 @@ public function getResponseJwt($resources)
...
     /**
-     * This method builds an auto-submitting HTML form to post the deep linking response message
-     * back to platform, as per LTI-DL 2.0 specification. The resulting HTML is then written to standard output,
-     * so calling this method will automatically send an HTTP response to conclude the content selection flow.
-     *
-     * @param  LtiDeepLinkResource[]  $resources The list of selected resources to be sent to the platform
-     *
-     * @todo Consider wrapping the content inside a well-formed HTML document,
-     * and returning it instead of directly writing to standard output
+     * @deprecated
```
Comment (dbhynds) [inline src/LtiDeepLink.php:47]: Once again, we shouldn't echo anything. Also... why are we building this form for people? They can't style it. If they need a form like this, they should build it themselves.

File: src/LtiGrade.php [hunk 4]
```diff
@@ -162,6 +162,8 @@ public function getCanvasExtension()
      *             is unexpected data. And, the type of LMS cannot simply be inferred by their URL.
      *
      * @see https://documentation.instructure.com/doc/api/score.html
+     *
+     * @todo: Deprecate this in the next major version
```
Comment (dbhynds) [inline src/LtiGrade.php:166]: I'm not certain about this one. Talking to Matt to see if this is something we actually want to do or not.

**Review by lin-brian-l — DISMISSED** [review]

File: src/LtiMessageLaunch.php [hunk 5]
```diff
@@ -153,9 +155,13 @@ public function initialize(array $request)
...
-        // @todo remove this in v6.0
-        if ($request === null) {
-            $request = $_POST;
+        // @TODO: Remove this on the next major release.
+        if (!isset($this->request)) {
+            if ($request === null) {
+                $request = $_POST;
+            }
+
+            $this->setRequest($request);
```
Comment (lin-brian-l) [inline src/LtiMessageLaunch.php:164]: `!isset()` can mean that the variable is `null` or `undefined`; does the case of `undefined` need to be addressed?

**Review by lin-brian-l — CHANGES_REQUESTED** [review]
@dbhynds great work! I have some minor comments but things look good so far.

File: .githooks/pre-commit [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+  fi
+}
+
+function skip_if_merge_in_progress {
+  if [ -f ".git/MERGE_HEAD" ]; then
+    echo "Detected merge in progress, skipping pre-commit hook." && exit 0
+  fi
+}
+
+function fail_if_on_master_branch {
```
[the remark this answers is not in the document]
Comment (dbhynds) [reply .githooks/pre-commit:77]: I don't think so, cos if someone forks the repo, they'd be working on the master branch of their fork

[the remark this answers is not in the document]
Comment (lin-brian-l) [reply .githooks/pre-commit:67]: @dbhynds sounds good, can this function be deleted if it's not being used then?

**Review by lin-brian-l — CHANGES_REQUESTED** [review]
@dbhynds I've got one last question about the githook!

**Review by lin-brian-l — DISMISSED** [review]

**Review by lin-brian-l — APPROVED** [review]

**Pull request merged**